statistics/update.go

```diff
@@ -453,17 +453,14 @@ func (h *Handle) HandleUpdateStats(is infoschema.InfoSchema) error {
 			}
 			continue
 		}
-		// dump the stats into kv
-		if hist != nil {
-			// Update the NDV of primary key column.
-			if col != nil && mysql.HasPriKeyFlag(col.Info.Flag) && PKIsHandle {
-				hist.NDV = int64(hist.totalRowCount())
-				col = nil
-			}
-			err = h.dumpStatsUpdateToKV(tableID, int(isIndex), q, hist, cms)
-			if err != nil {
-				return errors.Trace(err)
-			}
+		// Update the NDV of primary key column.
+		if col != nil && mysql.HasPriKeyFlag(col.Info.Flag) && PKIsHandle {
+			hist.NDV = int64(hist.totalRowCount())
+			col = nil
+		}
+		err = h.dumpStatsUpdateToKV(tableID, int(isIndex), histID, q, hist, cms)
```

Comment:
It will be initialized when we meet a new group, as indicated [here](https://github.com/pingcap/tidb/pull/7128/files#diff-43d226f85e4ecd4c612f6b737dd3ef61R466).

Reply:
If we group feedbacks in the first loop, then dump the feedbacks in the second loop, we can avoid the case that `histID` is `-1`.